GithubRepo/Scenes/Search/ViewModel/SearchViewModel.swift

```diff
@@ -6,50 +6,56 @@
 //
 
 import Foundation
+import RxRelay
+import RxSwift
 
 class SearchViewModel: ViewModelSearching {
     // MARK: - Attributes
 
+    private let disposeBag = DisposeBag()
     private var githubRepository: GithubRepository
-    private(set) var repositoryCellViewModels: [RepositoryCellViewModel]
-    private(set) var loading: Bool
-    private(set) var error: Bool
+    private(set) var repositoryCellViewModels: BehaviorRelay<[RepositoryCellViewModel]>
+    private(set) var state: BehaviorRelay<FetchState>
     private weak var coordinator: SearchCoordinator?
-    weak var delegate: SearchViewModelDelegate?
 
     // MARK: - Constructors
 
     init(coordinator: SearchCoordinator, repository: GithubRepository = GithubMainRepository()) {
         self.coordinator = coordinator
-        self.repositoryCellViewModels = []
-        self.loading = false
-        self.error = false
+        self.repositoryCellViewModels = BehaviorRelay(value: [])
+        self.state = BehaviorRelay(value: .inital)
         self.githubRepository = repository
-        self.githubRepository.delegate = self
+
+        self.bind()
     }
 
     // MARK: - Methods
 
     func fetchRepositories(query: String) {
         githubRepository.fetchRepositories(with: query)
     }
-}
-
-// MARK: - Notifications from Repository
 
-extension SearchViewModel: GithubRepositoryDelegate {
-    func didChangeLoading(loading: Bool) {
-        self.loading = loading
-        self.delegate?.onChangeSearchLoadingState(isLoading: loading)
+    private func bind() {
+        githubRepository.state.subscribe(onNext: {
+            self.onChangeState(state: $0)
+        })
```

Comment:
you can use more from RxSwift here. You can `bind` both observables instead of subscribing and calling the `accept` function.

as you are using relays here, you gonna use `drive` instead, but the concept is the same.

```suggestion
        githubRepository.state
            .drive(self.state)
```